### PR TITLE
Bumped bluepy 0.14.7/8, bluepysnap 0.4.1/0.5.0, neurom 1.4.20, libsonata 0.1.3

### PR DESCRIPTION
**Pull request by tomdele**

### CODE REVIEW

**Comment by tomdele** [comment]
Yo guys, 
I am updating few interdependent packages. 
@alkino can you have a look at the libsonata part ? just to check I have no done something wrong or not missed something. Thx.

**Review by matz-e — COMMENTED** [review]
From my side, all clear.

**Comment by alkino** [comment]
Wait for https://github.com/BlueBrain/spack/pull/782

**Comment by tomdele** [comment]
For the record :
we need an updated version of HighFive for libsonata==0.1.3. @alkino did the HighFive==2.2 version bump in the separated PR: #782

**Comment by alkino** [comment]
The feature needed in 2.2 is support of `enums` in `HighFive`

**Comment by tomdele** [comment]
Please retest

**Comment by matz-e** [comment]
> Please retest

did you rebase?

**Comment by tomdele** [comment]
yep indeed forgot to do it ... Thank you !

**Comment by jorblancoa** [comment]
retest this please jenkins

**Comment by jorblancoa** [comment]
This PR is blocked by Brion using and old version of HighFive (2.1.1) while libsonata needs the latest one (2.2.1).

**Comment by mgeplf** [comment]
> This PR is blocked by Brion using and old version of HighFive (2.1.1) while libsonata needs the latest one (2.2.1).

Any update on this @jorblancoa or @ppodhajski?

**Comment by ppodhajski** [comment]
We need a release of Brion to merge this PR: https://github.com/BlueBrain/spack/pull/812
@NadirRoGue  is about to finalize it.

**Comment by mgeplf** [comment]
Any update on this @jorblancoa or @ppodhajski?

**Comment by pramodk** [comment]
@tomdele : can you rebase on latest develop? #812 is now merged.

**Comment by tomdele** [comment]
@pramodk : Thanks a lot for sharing the info ! 
I did the rebase. Hope it works.

File: var/spack/repos/builtin/packages/libsonata/package.py [hunk 1]
```diff
@@ -26,8 +27,8 @@ class Libsonata(CMakePackage):
     depends_on('cmake@3.3:', type='build')
     depends_on('py-setuptools-scm', type='build', when='@0.1:')
     depends_on('fmt@4.0:')
-    depends_on('highfive+mpi', when='+mpi')
-    depends_on('highfive~mpi', when='~mpi')
+    depends_on('highfive+mpi@2.2.1', when='+mpi')
+    depends_on('highfive~mpi@2.2.1', when='~mpi')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/libsonata/package.py]: Can we make this version dependent? Pinning it like this is dangerous.

Comment (tomdele) [reply var/spack/repos/builtin/packages/libsonata/package.py]: Sure you are right. 

So you mean to : 
```
    depends_on('highfive+mpi@2.2.1:', when='+mpi')
    depends_on('highfive~mpi@2.2.1:', when='~mpi')
```
or something more fancy ?

Comment (tomdele) [reply var/spack/repos/builtin/packages/libsonata/package.py]: like :

```
depends_on('highfive+mpi', when='@:0.1.2+mpi')
depends_on('highfive~mpi', when='@:0.1.2~mpi')
depends_on('highfive+mpi@2.2.1:', when='@0.1.3:+mpi')
depends_on('highfive~mpi@2.2.1:', when='@0.1.3:~mpi')
```

Comment (matz-e) [reply var/spack/repos/builtin/packages/libsonata/package.py]: The latter. Given the error message, it looks like there needs to be a version pin somewhere else, too…

Comment (tomdele) [reply var/spack/repos/builtin/packages/libsonata/package.py]: Yeah everything is really entangled ...
Ok thx ! I will add this and I will bump the versions of the nse  packages also (we have new releases for bluepy and bluepysnap)

**Comment by jorblancoa** [comment]
Does brion needs to be released in version 3.1.0?https://github.com/BlueBrain/spack/blob/a20a40539e7e928092dc87e1b156c78417786b5d/deploy/packages/bbp-packages.yaml#L27

**Comment by tomdele** [comment]
At least not on my side.

**Comment by tomdele** [comment]
Should I bump ?

File: deploy/packages/bbp-packages.yaml [hunk 2]
```diff
@@ -120,7 +120,7 @@ packages:
       - nest@2.18.0
       - py-efel@3.0.80
       - py-bglibpy@4.2.23 ^neuron%intel@19.0.4
-      - py-bluepy@0.14.6
+      - py-bluepy@0.14.7
```
Comment (genric) [inline deploy/packages/bbp-packages.yaml]: 8?

Comment (tomdele) [reply deploy/packages/bbp-packages.yaml]: Yes I added it in the future commit but I did not push yet due to pb with building libsonata (not related to bluepy tho)

**Comment by jorblancoa** [comment]
> Should I bump ?

I think the CI is failing because of that...

**Comment by jorblancoa** [comment]
retest this please jenkins

**Review by jorblancoa — APPROVED** [review]

**Pull request merged**